process/economicMetrics.go

```diff
@@ -0,0 +1,70 @@
+package process
+
+import (
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+)
+
+// EconomicsDataPath represents the path where an observer exposes his economics data
+const EconomicsDataPath = "/network/economics"
+
+// GetEconomicsDataMetrics will return the economic metrics from cache
+func (nsp *NodeStatusProcessor) GetEconomicsDataMetrics() (*data.GenericAPIResponse, error) {
+	economicMetricsToReturn, err := nsp.economicMetricsCacher.Load()
+	if err == nil {
+		return economicMetricsToReturn, nil
+	}
+
+	log.Info("economic metrics: cannot get from cache. Will fetch from API", "error", err.Error())
+
+	return nsp.getEconomicsDataMetricsFromApi()
+}
+
+func (nsp *NodeStatusProcessor) getEconomicsDataMetricsFromApi() (*data.GenericAPIResponse, error) {
+	metaObservers, err := nsp.proc.GetObservers(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	return nsp.getEconomicsDataMetrics(metaObservers)
+}
+
+func (nsp *NodeStatusProcessor) getEconomicsDataMetrics(observers []*data.NodeData) (*data.GenericAPIResponse, error) {
+	for _, observer := range observers {
+		var responseNetworkMetrics *data.GenericAPIResponse
+
+		_, err := nsp.proc.CallGetRestEndPoint(observer.Address, EconomicsDataPath, &responseNetworkMetrics)
+		if err != nil {
+			log.Error("economics data request", "observer", observer.Address, "error", err.Error())
+			continue
+		}
+
+		log.Info("economics data request", "shard id", observer.ShardId, "observer", observer.Address)
+		return responseNetworkMetrics, nil
+	}
+
+	return nil, ErrSendingRequest
+}
+
+// StartCacheUpdate will update the economic metrics cache at a given time
+func (nsp *NodeStatusProcessor) StartCacheUpdate() {
+	go func() {
```

Comment:
we might add a guard variable here: initiated to 0 on L53, incremented on L57, reset on L61. if that value exceeds a certain threshold, store nil in the economicMetricsCacher

Reply:
done